gsrs-fda-substance-extension/src/main/java/fda/gsrs/substance/exporters/ClinicalTrialEuropeDTOExporter.java

```diff
@@ -0,0 +1,221 @@
+package fda.gsrs.substance.exporters;
+
+import gov.hhs.gsrs.clinicaltrial.europe.api.ClinicalTrialEuropeDTO;
+import gov.hhs.gsrs.clinicaltrial.europe.api.ClinicalTrialEuropeDrugDTO;
+import gov.hhs.gsrs.clinicaltrial.europe.api.ClinicalTrialEuropeMeddraDTO;
+import gov.hhs.gsrs.clinicaltrial.europe.api.ClinicalTrialEuropeProductDTO;
+import gsrs.module.substance.SubstanceEntityService;
+import ix.ginas.exporters.*;
+import ix.ginas.models.v1.Substance;
+
+import java.io.IOException;
+import java.util.*;
+
+enum ClinicalTrialEuropeDefaultColumns implements Column {
+    TRIAL_NUMBER,
+    TITLE,
+    SUBSTANCE_NAME,
+    SUBSTANCE_KEY,
+    CONDITIONS,
+    SPONSOR_NAME,
+    RESULTS
+    }
+
+public class ClinicalTrialEuropeDTOExporter implements Exporter<ClinicalTrialEuropeDTO> {
+
+    private final Spreadsheet spreadsheet;
+
+    private int row = 1;
+
+    private final List<ColumnValueRecipe<ClinicalTrialEuropeDTO>> recipeMap;
+
+    private static SubstanceEntityService substanceEntityService;
+
+    private ClinicalTrialEuropeDTOExporter(Builder builder, SubstanceEntityService substanceEntityService) {
+
+        this.substanceEntityService = substanceEntityService;
+
+        this.spreadsheet = builder.spreadsheet;
+        this.recipeMap = builder.columns;
+
+        int j = 0;
+        Spreadsheet.SpreadsheetRow header = spreadsheet.getRow(0);
+        for (ColumnValueRecipe<ClinicalTrialEuropeDTO> col : recipeMap) {
+            j += col.writeHeaderValues(header, j);
+        }
+    }
+
+    @Override
+    public void export(ClinicalTrialEuropeDTO s) throws IOException {
+        Spreadsheet.SpreadsheetRow row = spreadsheet.getRow(this.row++);
+
+        int j = 0;
+        for (ColumnValueRecipe<ClinicalTrialEuropeDTO> recipe : recipeMap) {
+            j += recipe.writeValuesFor(row, j, s);
+        }
+    }
+
+    @Override
+    public void close() throws IOException {
+        spreadsheet.close();
+    }
+
+    private static Map<Column, ColumnValueRecipe<ClinicalTrialEuropeDTO>> DEFAULT_RECIPE_MAP;
```

Comment:
I know this code is inspired by other exporters, but we need to rework this process. It was done as a hacky solution a long time ago and got canonized. It's probably okay for now but needs real attention.